ChangeOneSubscriptionOwnership/handler.ts

```diff
@@ -0,0 +1,149 @@
+    .table(dbConfig.DB_TABLE)
+    .update("status", status)
+    .from(dbConfig.DB_TABLE)
+    .where({ subscriptionId })
+    .toQuery() as NonEmptyString;
+
+/*
+ * The purpose of this function is to update a single subscription inside DB from a valid Update Query Statement
+ */
+export const updateSubscriptionStatusToDatabase = (
+  config: IConfig,
+  connect: Pool
+) => (
+  subscriptionId: NonEmptyString,
+  status: SubscriptionStatus
+): TE.TaskEither<IDbError, ResultSet> =>
+  pipe(
+    generateUpdateSQL(config)(subscriptionId, status),
+    sql => queryDataTable(connect, sql),
+    TE.mapLeft(flow(toPostgreSQLErrorMessage, toPostgreSQLError))
+  );
+
+/*
+ * The function purpose is to update the subscription owner id to APIM
+ * We need the complete target id PATH: /subscriptions/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX/resourceGroups/XXXXXXX/providers/Microsoft.ApiManagement/service/XXXXXXX/users/users/XXXXXXXXXXXXXXXXXXXXXXXXXX
+ */
+export const updateApimSubscription = (
+  config: IDecodableConfigAPIM,
+  apimClient: ApiManagementClient
+) => (
+  subscriptionId: NonEmptyString,
+  targetId: NonEmptyString
+): TE.TaskEither<IApimUserError, SubscriptionUpdateResponse> =>
+  pipe(
+    TE.tryCatch(
+      () =>
+        apimClient.subscription.update(
+          config.APIM_RESOURCE_GROUP,
+          config.APIM_SERVICE_NAME,
+          subscriptionId,
+          "*",
+          {
+            ownerId: targetId
+          }
+        ),
+      e => toApimUserError((e as Error).message)
+    ),
+    TE.chain(
+      flow(
+        TE.fromPredicate(
+          // Check if ownerId is available and it's the same of targetId
+          res => res.ownerId !== undefined && res.ownerId === targetId,
+          () => toApimUserError("Error on update")
+        )
+      )
+    )
+  );
+
+export const createHandler = (
+  config: IConfig,
+  apimClient: ApiManagementClient,
+  pool: Pool
+): Parameters<typeof withJsonInput>[0] =>
+  withJsonInput(
+    async (_context, subscriptionMessage): Promise<void> =>
+      // ToDo: Refactor mapLeft errors
+      pipe(
+        subscriptionMessage,
+        ClaimSubscriptionItem.decode,
+        TE.fromEither,
+        TE.mapLeft(() => E.toError("Error on decode")),
+        // Update subscription on APIM
+        TE.chain(subscriptionToMigrate =>
+          pipe(
+            updateApimSubscription(config, apimClient)(
+              subscriptionToMigrate.subscriptionId,
+              subscriptionToMigrate.targetId
+            ),
+            // TE.mapLeft(E.toError),
+            TE.orElseW(() =>
+              pipe(
+                updateSubscriptionStatusToDatabase(config, pool)(
+                  subscriptionToMigrate.subscriptionId,
+                  SubscriptionStatus.FAILED
+                ),
+                TE.mapLeft(() => E.toError(`Error on DB after APIM Fail`)),
+                TE.chainW(() => TE.left(E.toError("Error update on APIM")))
+              )
+            ),
```

Comment:
The meaning of this is:
> if subscription failed to update, then mark it as `FAILED` and set the handler to success.

Is it what you meant?

Reply:
Yes, you can get an error from an APIM update so you need to mark the subscription migration action as `FAILED`, but you can get also an error when you try to update the field on database. So `mapLeft` catch both errors.